assets/dev/js/editor/components/documents/commands/switch.js

```diff
@@ -3,9 +3,17 @@ export class Switch extends $e.modules.CommandBase {
 		this.requireArgument( 'id', args );
 	}
 
-	apply( args ) {
+	async apply( args ) {
 		const { id, mode, onClose, shouldScroll = true, setAsInitial = false } = args;
 
+		if ( ! elementor.documents.getCurrent().$element && elementor.config.document.panel.has_elements ) {
+			await new Promise( ( resolve ) => {
+				elementor.on( 'document:loaded', () => {
+					return resolve();
+				} );
+			} );
```

Comment:
Maybe the logic should be moved to the open/close commands (and also catch other race conditions):

```JS
// close.js

if ( args.document.isClosing ) {
    return Promise.reject( 'document is already being closed' );
}

if ( currentDocument.isOpening ) {
    await waitForDocumentToLoad( currentDocument.id );
}
```


```JS
// open.js

if ( args.document.isOpening ) {
    return Promise.reject( 'document is already being opened' );
}

if ( currentDocument.isClosing) {
    await waitForDocumentToUnload( currentDocument.id );
}
```